.buildkite/pipeline-nightly-e2e-tests.yml

```diff
@@ -0,0 +1,348 @@
+agents:
+  memory: "4G"
+
+steps:
+
+  - group: "aks"
+    steps:
+
+      - label: "aks"
+        commands:
+          - .ci/setenvconfig e2e/main
+          - .ci/setenvconfig e2e/aks
+          - .buildkite/scripts/common/get-test-artifacts.sh
+          - make run-deployer
+          - make e2e-run
+        artifact_paths:
+          - "eck-diagnostic-*.zip"
+
+      - wait: ~
+        continue_on_failure: true
+
+      - label: "cleanup"
+        commands:
+          - .ci/setenvconfig cleanup/aks
+          - make run-deployer
+
+  - group: "eks"
+    steps:
+
+      - label: "eks"
+        commands:
+          - .ci/setenvconfig e2e/main
+          - .ci/setenvconfig e2e/eks
+          - .buildkite/scripts/common/get-test-artifacts.sh
+          - make run-deployer
+          - make e2e-run
+        artifact_paths:
+          - "eck-diagnostic-*.zip"
+
+      - wait: ~
+        continue_on_failure: true
+
+      - label: "cleanup"
+        commands:
+          - .ci/setenvconfig cleanup/eks
+          - make run-deployer
+
+  - group: "eks-arm"
+    steps:
+
+      - label: "e2e/eks-arm"
+        commands:
+          - .ci/setenvconfig e2e/main
+          - .ci/setenvconfig e2e/eks-arm
+          - .buildkite/scripts/common/get-test-artifacts.sh
+          - make run-deployer
+          - make e2e-run
+        artifact_paths:
+          - "eck-diagnostic-*.zip"
```

Comment:
Forgive the dumb questions, but are all these groups and steps run in parallel?  And if so, are the `artifact_paths` being the same going to cause any collision?  Should they be named indicative of the test name they're running?

Reply:
Yes, groups and steps run in parallel. Only cleanup steps wait for all creation steps to have completed before running using the [wait step](https://buildkite.com/docs/pipelines/wait-step) . Each step has its own `artifact_paths` so there is no risk of collision.

![2022-10-04-103001_630x838_scrot](https://user-images.githubusercontent.com/582883/193775289-9ec6c9eb-9258-4b86-801d-9d81c8acd777.png)